File: Data Structures & Algorithms/insert-new-interval/submission-3.cpp
class Solution {
   public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> ans;
        int idx = 0;
        while (idx < intervals.size() and intervals[idx][1] < newInterval[0]) {
            ans.emplace_back(intervals[idx]);
            ++idx;
        }
        while (idx < intervals.size() and newInterval[1] >= intervals[idx][0]) {
            newInterval[0] = min(intervals[idx][0], newInterval[0]);
            newInterval[1] = max(intervals[idx][1], newInterval[1]);
            ++idx;
        }
        ans.emplace_back(newInterval);
        while (idx < intervals.size()) {
            ans.emplace_back(intervals[idx]);
            ++idx;
        }
        return ans;
    }
};

File: Data Structures & Algorithms/non-overlapping-intervals/submission-3.py
class Solution:
    def dfs(self, intervals, cidx,lidx):
        if cidx==len(intervals):
            return 0
        ans = self.dfs(intervals,cidx+1, lidx)
        if lidx==-1 or intervals[cidx][0] >=intervals[lidx][1]:
            ans = max(ans, 1 + self.dfs(intervals,cidx+1,cidx))

        return ans 


    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort()
        return len(intervals) - self.dfs(intervals,0,-1)